src/util.ts

```diff
@@ -627,3 +629,30 @@ export async function normalizeAndVerifySourceDir(sourceDir: string): Promise<st
 export function isCodespaces(): boolean {
   return !!process.env["CODESPACES"];
 }
+
+export function getAllFilePaths(dir: vscode.Uri, filename: string): string[] | undefined {
+  const regex: RegExp = new RegExp(`(\\/|\\\\)${escapeStringRegExp(filename)}$`);
+  return recGetAllFilePaths(dir.fsPath, filename, regex, fs.readdirSync(dir.fsPath), []);
+}
+
+function recGetAllFilePaths(dir: string, filename: string, regex: RegExp, files: string[], result: string[]) {
+  for (const item of files) {
+    const file = path.join(dir, item);
+    if (fs.statSync(file).isDirectory()) {
```

Comment:
>fs.statSync(file) [](http://example.com/codeflow?start=8&length=17)

You should save the result of this `fs.statSync` in a variable so that you don't have to call it twice (for the isFile case)